controller/setting_controller.go

```diff
@@ -390,6 +395,85 @@ func (sc *SettingController) syncBackupTarget() (err error) {
 	return nil
 }
 
+func (sc *SettingController) handleSecretsForAWSIAMRoleAnnotation(backupTargetURL, oldSecretName, newSecretName string, isBackupTargetURLChanged bool) (err error) {
+	isSameSecretName := oldSecretName == newSecretName
+	if isSameSecretName && !isBackupTargetURLChanged {
+		return nil
+	}
+
+	isHavingAWSIAMRoleArn := false
+	if !isSameSecretName {
+		isHavingAWSIAMRoleArn, _, err = sc.updateSecretForAWSIAMRoleAnnotation(backupTargetURL, oldSecretName, true)
+		if err != nil {
+			return err
+		}
+	}
+	_, isNewSecretNotExist, err := sc.updateSecretForAWSIAMRoleAnnotation(backupTargetURL, newSecretName, false)
+	if err != nil {
+		return err
+	}
+	// kubernetes_secret_controller will not reconcile the secret that does not exist such as named "", so we remove AWS IAM role annotation of pods if new secret name is "".
+	if isNewSecretNotExist && isHavingAWSIAMRoleArn {
+		if err = sc.removePodsAWSIAMRoleAnnotation(); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (sc *SettingController) updateSecretForAWSIAMRoleAnnotation(backupTargetURL, secretName string, isOldSecret bool) (bool, bool, error) {
```

Comment:
nit: `(bool, bool, error)` can be difficult to comprehend at first glance, consider assigning variables or adding a comment.

Reply:
Added a comment and assigned variables. Thanks